### PR TITLE
Update sdk & minor changes to connection of cluster

### LINKED ISSUE #9702: [Bug]: Fix Couchbase
### Short description of current behavior

Currently in couchbase integration with MindsDB, there are a few minor bugs, related to SDK connecting to the cluster, which needs to be changed. Also the sdk needs to be upgraded.

### Video or screenshots

_No response_

### Expected behavior

_No response_

### How to reproduce the error

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by prajwal-pai77**
## Description

- Updates the python couchbase sdk to latest version.
- Updated the connection to couchbase by using connection_string.
- Updated relevant docs.

Fixes #9702 

## Type of change

(Please delete options that are not relevant)

- 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [X] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Review by MinuraPunchihewa — COMMENTED** [review]
Hey @prajwal-pai77,
Thank you for this PR.
Will the protocol for the connection string always be `couchbase://`? In that case, I think we would prefer the `host` parameter; we can then build the connection string in the connection logic. Even if this is not so, I think we can maybe use an optional `protocol` parameter along with `host`. It can default to the most commonly used protocol (e.g. `couchbase`). Our [Clickhouse integration](https://docs.mindsdb.com/integrations/data-integrations/clickhouse) follows a similar pattern.

Can you also please confirm that you do not run into any dependency conflicts after updating the version of the SDK?

Finally, it looks like you've got a small lint issue that our CI has detected.

**Comment by prajwal-pai77** [comment]
Hey @MinuraPunchihewa 
Thanks for taking a look.
The connection string will not always be couchbase://, When using couchbase capella, the `connection_string` can be found out in the connect tab, and it will contain the couchbases://, so just specifying that should be enough. I have mentioned it in the doc:
```the `connection_string` can be found out when you navigate under the Connect tab.```
When running locally, couchbase:// is sufficient as shown in the example.

I have tested the couchbase_handler.py file after upgrading the sdk, please let me know if i need to test anything else, so i don't run into dependency conflicts.

How can i fix this lint issue? Is there anthing i can refer to?

File: docs/integrations/data-integrations/couchbase.mdx [hunk 1]
```diff
@@ -20,14 +20,16 @@ Before proceeding, ensure the following prerequisites are met:
...
 
 The required arguments to establish a connection are as follows:
-* `host`: the host name or IP address of the Couchbase server
+* `connection_string`: the Connection string to specify the cluster endpoint
 * `bucket`: the bucket name to use when connecting with the Couchbase server
 * `user`: the user to authenticate with the Couchbase server
 * `password`: the password to authenticate the user with the Couchbase server
 * `scope`:  scopes are a level of data organization within a bucket. If omitted, will default to `_default`
 
+Note: The connection string expects the couchbases:// or couchbase:// part.
```
Comment (MinuraPunchihewa) [inline docs/integrations/data-integrations/couchbase.mdx:29]: Let's make this,
Note: The connection string expects either the couchbases:// or couchbase:// protocol.

Please make the same change in the README.

File: docs/integrations/data-integrations/couchbase.mdx [hunk 2]
```diff
@@ -20,14 +20,16 @@ Before proceeding, ensure the following prerequisites are met:
...
+
 <Tip>
-If you are using Couchbase Capella, the `host` should be the connection string for the cluster. When you navigate to your cluster, the connection string can be found under the Connect tab.
+If you are using Couchbase Capella, the `connection_string` can be found out when you navigate under the Connect tab.
```
Comment (MinuraPunchihewa) [inline docs/integrations/data-integrations/couchbase.mdx:32]: Let's make this:
If you are using Couchbase Capella, you can find the connection_string under the Connect tab.

Please make the same change in the README.

File: docs/integrations/data-integrations/couchbase.mdx [hunk 3]
```diff
@@ -20,14 +20,16 @@ Before proceeding, ensure the following prerequisites are met:
 This handler is implemented using the `couchbase` library, the Python driver for Couchbase.
 
 The required arguments to establish a connection are as follows:
-* `host`: the host name or IP address of the Couchbase server
+* `connection_string`: the Connection string to specify the cluster endpoint
```
Comment (MinuraPunchihewa) [inline docs/integrations/data-integrations/couchbase.mdx:23]: Let's make this,
the connection string for the endpoint of the Couchbase server

Please make the same change in the README.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
I left a few minor comments to fix the grammar in the docs.

In order to fix the CI, simply remove the unused import of `urllib.parse.urlparse` in `couchbase_handler.py.`

**Review by MinuraPunchihewa — APPROVED** [review]
LGTM!

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb/blob/main/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by ZoranPandovski** [comment]
@prajwal-pai77 Can you please sign the CLA?

**Comment by prajwal-pai77** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**